yarn-project/simulator/src/public/abstract_phase_manager.ts

```diff
@@ -350,7 +352,27 @@ export abstract class AbstractPhaseManager {
     // TODO(#3675): This should be done in a public kernel circuit
     removeRedundantPublicDataWrites(kernelOutput, this.phase);
 
-    return [publicKernelInputs, kernelOutput, kernelProof, newUnencryptedFunctionLogs, undefined, returns, gasUsed];
+    return [
+      publicKernelInputs,
+      kernelOutput,
+      kernelProof,
+      newUnencryptedFunctionLogs,
+      undefined,
+      enqueuedCallResults,
+      gasUsed,
+    ];
+  }
+
+  /**
+   * Recursively accummulate the return values of a public call result and its nested executions,
+   * so they can be retrieved in order.
+   * @param executionResult
+   * @returns
+   */
+  #accumulateReturnValues(executionResult: PublicExecutionResult): NestedProcessReturnValues {
```

Comment:
Should we have a pointer to the other place where the was something very much like this? The pxe_service?

Reply:
Yeah, functions were initially very different but ended up practivally the same barring types...I'll look for somewhere to put a common version

Reply:
Done!